diff --git a/variety/data/scripts/set_wallpaper b/variety/data/scripts/set_wallpaper
index 98c88e4..4f231b2 100755
--- a/variety/data/scripts/set_wallpaper
+++ b/variety/data/scripts/set_wallpaper
@@ -259,7 +259,7 @@ elif [ "$DE" == "fluxbox" ]; then
     mkdir -p ~/.fluxbox
     echo '$''full ''$'"full|$WP||:0.0" >~/.fluxbox/lastwallpaper
 
-elif [ "$DE" == "sway" ]; then
+elif true; then
     # If swaybg is available, use it as prevents system freeze.
     # See https://github.com/swaywm/sway/issues/5606
     if command -v "swaybg" >/dev/null 2>&1; then
