diff --git a/variety/data/scripts/set_wallpaper b/variety/data/scripts/set_wallpaper
index 98c88e4..515ad03 100755
--- a/variety/data/scripts/set_wallpaper
+++ b/variety/data/scripts/set_wallpaper
@@ -36,6 +36,8 @@
 # The default is to simply set WP=$1.
 WP=$1
 SCRIPT_PATH="$(readlink -f "$0")"
+rm ~/.background-image
+ln -s "$WP" ~/.background-image
 
 detect_desktop() {
     if [ -n "$XDG_CURRENT_DESKTOP" ]; then
