diff --git a/src/components/bar/modules/workspaces/helpers/utils.ts b/src/components/bar/modules/workspaces/helpers/utils.ts
index 5d2c033..3ce3647 100644
--- a/src/components/bar/modules/workspaces/helpers/utils.ts
+++ b/src/components/bar/modules/workspaces/helpers/utils.ts
@@ -251,7 +251,7 @@ export const renderLabel = (
     monitor: number,
 ): string => {
     if (showAppIcons) {
-        return appIcons;
+        return (workspaceMask ? `${index + 1}` : `${i}`) + " " + appIcons;
     }
 
     if (showIcons) {
