diff --git a/src/components/bar/modules/kblayout/index.tsx b/src/components/bar/modules/kblayout/index.tsx
index ca4800f..7218d10 100644
--- a/src/components/bar/modules/kblayout/index.tsx
+++ b/src/components/bar/modules/kblayout/index.tsx
@@ -14,10 +14,25 @@ const hyprlandService = AstalHyprland.get_default();
 const { label, labelType, icon, leftClick, rightClick, middleClick, scrollUp, scrollDown } =
     options.bar.customModules.kbLayout;
 
+function replaceLangWithIcon(lang: String): String {
+    switch (lang) {
+        case "US":
+            return "🇺🇸";
+        case "CZ (Qwerty)":
+            return "🇨🇿"
+        case "UA":
+            return "🇺🇦"
+        case "RU":
+            return "💩"
+        default:
+            return lang;
+    }
+}
+
 function setLabel(self: Astal.Label): void {
     try {
         const devices = hyprlandService.message('j/devices');
-        self.label = getKeyboardLayout(devices, labelType.get());
+        self.label = replaceLangWithIcon(getKeyboardLayout(devices, labelType.get()));
     } catch (error) {
         console.error(error);
     }
